packages/indexer-agent/src/query-fees/allocations.ts

```diff
@@ -194,14 +203,104 @@ export class AllocationReceiptCollector implements ReceiptCollector {
   private startVoucherProcessing() {
     timer(30_000).pipe(async () => {
       const pendingVouchers = await this.pendingVouchers()
-      for (const voucher of pendingVouchers) {
-        await this.submitVoucher(voucher)
+
+      const logger = this.logger.child({})
+
+      const vouchers = await pReduce(
+        pendingVouchers,
+        async (results, voucher) => {
+          if (
+            await this.allocationExchange.allocationsRedeemed(
+              voucher.allocation,
+            )
+          ) {
+            logger.warn(
+              `Query fee voucher for allocation already redeemed, deleted local voucher copy`,
+              { allocation: voucher.allocation },
+            )
+            try {
+              await this.models.vouchers.destroy({
+                where: { allocation: voucher.allocation },
+              })
+            } catch (err) {
+              logger.warn(
+                `Failed to delete local vouchers copy, will try again later`,
+                { err, allocation: voucher.allocation },
+              )
+            }
+            return results
+          }
+          if (
+            BigNumber.from(voucher.amount).lt(this.voucherRedemptionThreshold)
+          ) {
+            results.belowThreshold.push(voucher)
+          } else {
+            results.eligible.push(voucher)
+          }
+          return results
+        },
+        { belowThreshold: <Voucher[]>[], eligible: <Voucher[]>[] },
+      )
+
+      if (vouchers.belowThreshold.length > 0) {
+        logger.info(`Query vouchers below the redemption threshold`, {
+          hint: 'If you would like to redeem vouchers like this, reduce the voucher redemption threshold',
+          voucherRedemptionThreshold: formatGRT(
+            this.voucherRedemptionThreshold,
+          ),
+          belowThresholdCount: vouchers.belowThreshold.length,
+          totalValueGRT: formatGRT(
+            vouchers.belowThreshold.reduce(
+              (total, voucher) => total.add(BigNumber.from(voucher.amount)),
+              BigNumber.from(0),
+            ),
+          ),
+          allocations: vouchers.belowThreshold.map(
+            voucher => voucher.allocation,
+          ),
+        })
+      }
+
+      // If there are no eligible vouchers then bail
+      if (vouchers.eligible.length === 0) return
+
+      const voucherBatch = vouchers.eligible.slice(
```

Comment:
Sort before slice?

Reply:
O I see now that the `pendingVouchers()` function uses a sorted query; would be useful to note that in a comment here for clarity.

Reply:
Will do